### PR TITLE
[7103] Test for iticket create with relative paths (main)

### PR DESCRIPTION
**Pull request by Nishant2022**
Connected to https://github.com/irods/irods_client_icommands/pull/395

### CODE REVIEW

**Comment by Nishant2022** [comment]
See https://github.com/irods/irods_client_icommands/pull/395#issuecomment-1619051671

File: scripts/irods/test/test_iticket.py [hunk 1]
```diff
@@ -766,3 +766,47 @@ def test_write_tickets_for_collections_allow_creation_of_data_objects__issue_591
...
+        collection_2, collection_2_ticket_name = 'issue_7103_collection_2', 'collection_2_ticket'
+        data_object_1, data_object_1_ticket_name = 'issue_7103_data_object_1', 'data_object_1_ticket'
+        data_object_2, data_object_2_ticket_name = 'issue_7103_data_object_2', 'data_object_2_ticket'
+        home_data_object, home_data_object_ticket_name = 'issue_7103_home_data_object', 'home_data_object_ticket'
+
+        home_collection_ticket_name = 'home_collection_ticket'
+
+        # Create collections
+        self.admin.assert_icommand(['imkdir', collection_1])
+        self.admin.assert_icommand(['imkdir', collection_2])
```
Comment (alanking) [inline scripts/irods/test/test_iticket.py:781]: Is this test class cleaning up the collections/data objects after it's done? Even when it fails?

Consider using the "session collection" for the active user which will always be cleaned up by the test framework. If that's not feasible due to the relative nature of the paths in the test, let's wrap the test with a try/finally block which cleans up the collections afterwards.

File: scripts/irods/test/test_iticket.py [hunk 2]
```diff
@@ -766,3 +766,47 @@ def test_write_tickets_for_collections_allow_creation_of_data_objects__issue_591
...
+        # Create objects
+        self.admin.assert_icommand(['itouch', f'{collection_1}/{data_object_1}'])
+        self.admin.assert_icommand(['itouch', f'{collection_2}/{data_object_2}'])
+        self.admin.assert_icommand(['itouch', home_data_object])
+
+        self.admin.assert_icommand(['icd', collection_1])
+
+        # Create ticket for current working directory
+        self.admin.assert_icommand(['iticket', 'create', 'read', '.', collection_1_ticket_name])
+        self.admin.assert_icommand(['iticket', 'ls', collection_1_ticket_name], 'STDOUT', [collection_1_ticket_name])
```
Comment (alanking) [inline scripts/irods/test/test_iticket.py:792]: Should a different user use the tickets in this test? Should we run the tests as a rodsuser?

File: scripts/irods/test/test_iticket.py (hunk 1 above)
Comment (Nishant2022) [reply scripts/irods/test/test_iticket.py:781]: I believe it is necessary to create some collections in order to demonstrate possible relative paths. I will add a try/finally block to remove the collections when done.

File: scripts/irods/test/test_iticket.py (hunk 2 above)
Comment (alanking) [reply scripts/irods/test/test_iticket.py]: Okay. Admin-only, full-permissions tests have bitten us in the past which is why I bring it up.

File: scripts/irods/test/test_iticket.py [hunk 3]
```diff
@@ -767,14 +767,68 @@ def test_write_tickets_for_collections_allow_creation_of_data_objects__issue_591
...
+        collection_1, collection_1_ticket_name = 'user_issue_7103_collection_1', 'user_coll_1_ticket'
+        collection_2, collection_2_ticket_name = 'user_issue_7103_collection_2', 'user_coll_2_ticket'
+        data_object_1, data_object_1_ticket_name = 'user_issue_7103_data_object_1', 'user_object_1_ticket'
+        data_object_2, data_object_2_ticket_name = 'user_issue_7103_data_object_2', 'user_object_2_ticket'
+        home_data_object, home_data_object_ticket_name = 'user_issue_7103_home_data_object', 'user_home_object_ticket'
+
+        home_collection_ticket_name = 'user_home_coll_ticket'
 
         try:
             self.admin.assert_icommand(['icd'])
```
[the remark this answers is not in the document]
Comment (Nishant2022) [reply scripts/irods/test/test_iticket.py]: Yeah I realized as soon as I pushed the commit. I pushed a second commit to fix it.

File: scripts/irods/test/test_iticket.py [hunk 4]
```diff
@@ -767,14 +767,68 @@ def test_write_tickets_for_collections_allow_creation_of_data_objects__issue_591
...
+            # Create ticket for sibling directory
+            self.admin.assert_icommand(['iticket', 'create', 'read', f'../{collection_2}', collection_2_ticket_name])
+            self.admin.assert_icommand(['iticket', 'ls', collection_2_ticket_name], 'STDOUT', [collection_2_ticket_name])
+
+            # Create ticket for object in sibling directory
+            self.admin.assert_icommand(['iticket', 'create', 'read', f'../{collection_2}/{data_object_2}', data_object_2_ticket_name])
+            self.admin.assert_icommand(['iticket', 'ls', data_object_2_ticket_name], 'STDOUT', [data_object_2_ticket_name])
+
+        finally:
+            # Cleanup
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply scripts/irods/test/test_iticket.py:818]: Let's clean them up. It just decreases the chance of running into errors.

[the remark this answers is not in the document]
Comment (Nishant2022) [reply scripts/irods/test/test_iticket.py:818]: Added cleanup

File: scripts/irods/test/test_iticket.py [hunk 5]
```diff
@@ -820,6 +820,13 @@ def test_ticket_create_with_relative_paths_admin__issue_7103(self):
             self.admin.assert_icommand(['irm', '-rf', collection_1])
             self.admin.assert_icommand(['irm', '-rf', collection_2])
             self.admin.assert_icommand(['irm', '-f', home_data_object])
+            
+            self.admin.assert_icommand(['iticket', 'delete', collection_1_ticket_name])
```
Comment (alanking) [inline scripts/irods/test/test_iticket.py:824]: Might want to change `assert_icommand` to `run_icommand` just in case one fails.

**Review by alanking — COMMENTED** [review]
I think I only had one more thing and I think this is good to go

**Review by alanking — APPROVED** [review]
Very nice. Squash em and please leave off #s, and we'll have one last look

**Review by alanking — APPROVED** [review]
Looks good to me. Let's wait for one more approval and then add #s

**Review by korydraughn — APPROVED** [review]
Let's add the pounds.

**Pull request merged**